presto-hive/src/main/java/com/facebook/presto/hive/S3SelectRecordCursorProvider.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.RecordCursor;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.joda.time.DateTimeZone;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_FILESYSTEM_ERROR;
+
+public class S3SelectRecordCursorProvider
+        implements HiveRecordCursorProvider
+{
+    private static final Logger log = Logger.get(S3SelectRecordCursorProvider.class);
+
+    public static final String S3_SELECT_OPTIMIZED = "s3.select.optimized";
+
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HiveClientConfig clientConfig;
+
+    private static Set<String> csvSerdes = ImmutableSet.of(
+            "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe");
+
+    @Inject
+    public S3SelectRecordCursorProvider(HdfsEnvironment hdfsEnvironment, HiveClientConfig clientConfig)
+    {
+        this.hdfsEnvironment = hdfsEnvironment;
+        this.clientConfig = clientConfig;
+    }
+
+    @Override
+    public Optional<RecordCursor> createRecordCursor(Configuration configuration, ConnectorSession session, Path path, long start, long length, long fileSize, Properties schema, List<HiveColumnHandle> columns, TupleDomain<HiveColumnHandle> effectivePredicate, DateTimeZone hiveStorageTimeZone, TypeManager typeManager)
+    {
+        boolean isOptimizedForS3 = configuration.getBoolean(S3_SELECT_OPTIMIZED, false);
+        if (!isOptimizedForS3) {
+            log.info("S3Select is not used for path: %s", path);
```

Comment:
Do we really need all these log statements?

Reply:
Yes this one is required because if for any reason the optimizer doesn't get enabled (unsupported compression codec identified in BackgroundHiveSplitLoader) then this is where one could check that even though they enabled the connecter it was tuned off. 
I changed this to debug.